packages/ide-extension/src/types/telemetry.ts

```diff
@@ -73,3 +74,63 @@ export interface TelemetryUIGoToPreviousPage extends TelemetryUIEventProps {
     nodeIdPath: string;
     nodeLevel: string;
 }
+
+export interface TelemetryUISelectOutcomeProps extends TelemetryUIEventProps {
+    action: 'SELECT_OUTCOME';
+    solved: string;
+    treeId: string;
+    treeTitle: string;
+    lastNodeId: string;
+    lastNodeTitle: string;
+    nodeIdPath: string;
+    nodeLevel: string;
+}
+
+export interface TelemetryUICommentProps extends TelemetryUIEventProps {
+    action: 'COMMENT';
+    treeId: string;
+    treeTitle: string;
+    lastNodeId: string;
+    lastNodeTitle: string;
+    nodeIdPath: string;
+    nodeLevel: string;
+}
+
+export interface TelemetryUISearchProps extends TelemetryUIEventProps {
+    action: 'SEARCH';
+    treeCount: string;
+    productFilterCount: string;
+    componentFilterCount: string;
+}
+
+export interface TelemetryUIExecuteCommandProps extends TelemetryUIEventProps {
+    action: 'EXECUTE_COMMAND';
+    commandLabel: string;
+    treeId: string;
+    treeTitle: string;
+    lastNodeId: string;
+    lastNodeTitle: string;
+    nodeIdPath: string;
+    nodeLevel: string;
+}
+
+export interface TelemetryUIFilterComponentsProps extends TelemetryUIEventProps {
+    action: 'FILTER_COMPONENTS';
+}
+
+export interface TelemetryUIFilterProductsProps extends TelemetryUIEventProps {
+    action: 'FILTER_PRODUCTS';
+}
+
+export interface TelemetryUIOpenLinkProps extends TelemetryUIEventProps {
+    action: 'OPEN_LINK';
+}
+
+export interface TelemetryUIShareLinkProps extends TelemetryUIEventProps {
+    action: 'SHARE_LINK';
+}
+
+export interface TelemetryUIClearFiltersProps extends TelemetryUIEventProps {
+    action: 'CLEAR_FILTERS';
+    justOpened: string;
```

Comment:
@oliversberzs, while reviewing this PR I saw what is required to fill this, including changes to multiple reducers. I'd propose we remove property `justOpened`. With your change to use a uuid it should be easy enough to track whether STARTUP and CLEAR_FILTERS was called subsequently or whether multiple steps have been in between.